pkg/operator/staticpod/flock.go

```diff
@@ -0,0 +1,95 @@
+package staticpod
+
+import (
+	"os"
+	"sync"
+	"syscall"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// FLock a type that supports file locking to coordinate work between processes
+type FLock struct {
+	locker sync.Mutex
+
+	lockFilePath string
+	lockedFile   *os.File
+}
+
+// NewFLock instantiates *FLock on the given path.
+// The path must point to a file not a directory.
+// The file doesn't have to exist prior to calling this method.
+// It will be creating on the first call to TryLock method.
+func NewFLock(filePath string) *FLock {
+	return &FLock{locker: sync.Mutex{}, lockFilePath: filePath}
+}
+
+// TryLock tries to acquire an exclusive lock on a file
+// until it succeeds, an error is returned or the timeout is reached.
+//
+// The callers MUST call the corresponding Unlock method to release the lock and associated resources except when an error
+// is returned (including the timeout).
+//
+// This method is safe for concurrent access.
+//
+// Note the given timeout shouldn't be less than 1 second.
+func (f *FLock) TryLock(timeout time.Duration) error {
```

Comment:
not a context?

Reply:
we could but then the usage becomes
```
ctx, cancel := context.WithTimeout()
defer cancel()
flock.TryLock(ctx)
```

vs

`flock.TryLock(time.Second * 5)`